### PR TITLE
feat: Trigger Node

### LINKED ISSUE #1437: Trigger node label, trigger type, trigger name changing
Addressing https://github.com/microsoft/BotFramework-Composer/issues/1428, https://github.com/microsoft/BotFramework-Composer/issues/1244, https://github.com/microsoft/BotFramework-Composer/issues/1407, and an additional interaction design issue that appeared through some of the decisions we made separately.

Currently, once you create any type of trigger, you can change its name on the right pane. I don't think this should be possible at this point since we are not allowing users to define names for triggers, besides the intent trigger(even for the intent trigger, we ask users to specify 'intent name' which we are making it its trigger name as well). 

My suggestions regarding this:
- The re-naming behavior on the right pane should be removed for all triggers except the intent trigger.
- For the intent trigger, it is not clear whether the name you specified in the trigger wizard is the name for your intent or trigger – because you were just asked to fill out an intent name but what you are changing on the right pane is a trigger name. It should be **the trigger name** that users would be asked to specify in the trigger wizard, not the intent name.

Regarding #1428, here's my design suggestion (see the designs in the red box)
https://www.figma.com/file/04DymSq4xElhOBdAyutvvTbq/Composer-SH-06.19?node-id=3054%3A195

### PR DESCRIPTION
**Pull request by yeze322**
## Description

image

## Task Item

closes #1437 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code refactor (non-breaking change which improve code quality, clean up, add tests, etc)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Doc update (document update)

## Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have functionally tested my change

## Screenshots

Please include screenshots or gifs if your PR include UX changes.

### CODE REVIEW

**Comment by yeze322** [comment]
We need to finalize what's the exact content (grey text) shown in trigger text of each trigger type
image

Collected some details from design:

|Trigger Type|Grey text to show|
|-------|--------|
|Intent| `(Intent trigger)`|
|Unrecognized intent|`(Unrecognized intent trigger)`|
|Dialog trigger|?|
|Activity trigger|?|
|Message activity trigger|?|
|Custom trigger|?|

(Obviously, the trigger name in black should be `$designer.name` or the same as grey text)

**Comment by mareekuh** [comment]
Please show the sub-trigger type name vs the overarching trigger : this is not only more informative, it also removed the repeat of the word 'trigger'.
So for user input this would be Intent or 'not recognized intent'
And for Dialog Triggers

- Dialog started (BeginDialog)
- Dialog cancelled (CancelDialog)
- Error occurred
- Re-prompt for input

For Activity Trigger:

- Conversation started (ConversationUpdate)
- Conversation ended
- Handover to human
- Conversation invoked
- User is typing
- Event received

and for message activity:

- Message received
- Message updated
- Message deleted
- Message reaction (thumbs up/down)

Custom trigger can remain Custom trigger.

**Comment by yeze322** [comment]
fixed broken test cases

**Comment by cwhitten** [comment]
@yeze322 we should roll up the form changes into this PR.

File: Composer/packages/extensions/visual-designer/src/components/nodes/Trigger.tsx [hunk 1]
```diff
@@ -5,42 +5,70 @@
...
-        themeColor: '#BFEAE9',
-        iconColor: 'black',
+function getName(data: any): string {
+  if (data && data.$designer && data.$designer.name) {
+    return data.$designer.name;
+  }
+  return getLabel(data);
+}
+
+const titleStyle: any = {
```
Comment (cwhitten) [inline Composer/packages/extensions/visual-designer/src/components/nodes/Trigger.tsx]: these should be defined in the styles file

File: Composer/packages/extensions/visual-designer/src/components/nodes/Trigger.tsx [hunk 2]
```diff
@@ -5,42 +5,70 @@
...
+  whiteSpace: 'nowrap',
+  color: '#333333',
+  fontFamily: 'Segoe UI',
+  fontSize: '18px',
+  lineHeight: '24px',
+};
+
+const subtitleStyle: any = {
```
Comment (cwhitten) [inline Composer/packages/extensions/visual-designer/src/components/nodes/Trigger.tsx]: move to styles file

File: Composer/packages/extensions/visual-designer/src/components/nodes/Trigger.tsx [hunk 3]
```diff
@@ -5,42 +5,70 @@
...
+};
+
+export const Trigger = ({ data, onClick = () => {} }): JSX.Element => {
+  const name = getName(data);
+  const label = getLabel(data);
+
+  return (
+    <div
+      className="Trigger--Container"
+      css={{
```
Comment (cwhitten) [inline Composer/packages/extensions/visual-designer/src/components/nodes/Trigger.tsx]: move to styles file

File: Composer/packages/extensions/visual-designer/src/components/nodes/Trigger.tsx [hunk 4]
```diff
@@ -5,42 +5,70 @@
...
-      header={'Trigger'}
-      label={getLabel(data)}
-      onClick={onClick}
-    />
-  </div>
-);
+    >
+      <div
+        className="Trigger--Content"
+        css={{ wordBreak: 'break-all', display: 'flex', flexDirection: 'column', alignItems: 'center' }}
```
Comment (cwhitten) [inline Composer/packages/extensions/visual-designer/src/components/nodes/Trigger.tsx]: move to styles file

File: Composer/packages/extensions/visual-designer/src/components/nodes/Trigger.tsx [hunk 5]
```diff
@@ -5,42 +5,70 @@
...
-  >
-    <FormCard
-      nodeColors={{
-        themeColor: '#BFEAE9',
-        iconColor: 'black',
+function getName(data: any): string {
+  if (data && data.$designer && data.$designer.name) {
+    return data.$designer.name;
+  }
+  return getLabel(data);
```
Comment (a-b-r-o-w-n) [inline Composer/packages/extensions/visual-designer/src/components/nodes/Trigger.tsx]:
```ts
return get(data, '$designer.name', getLabel(data));
```

File: Composer/packages/extensions/visual-designer/src/components/nodes/Trigger.tsx [hunk 6]
```diff
@@ -5,42 +5,70 @@
...
+    >
+      <div
+        className="Trigger--Content"
+        css={{ wordBreak: 'break-all', display: 'flex', flexDirection: 'column', alignItems: 'center' }}
+      >
+        <div className="Trigger--Title" css={titleStyle}>
+          <Icon iconName="Flow" style={{ lineHeight: '24px', marginRight: '5px' }} />
+          <span>{name}</span>
+        </div>
+        <div className="Trigger--Subtitle" css={subtitleStyle}>
```
Comment (a-b-r-o-w-n) [inline Composer/packages/extensions/visual-designer/src/components/nodes/Trigger.tsx]: Are these classnames being used?

**Comment by yeze322** [comment]
@a-b-r-o-w-n @alanlong9278 could you help rolling up the Form part's changes? I'm not quite familiar with it, thanks!

**Comment by yeze322** [comment]
Thanks Chris for updating those titles, let me fix CI failures.

**Pull request merged**